faunadb-common/src/main/java/com/faunadb/common/Connection.java

```diff
@@ -27,10 +31,21 @@
  * for the underlying implementation.
  */
 
-public class Connection {
-  static final int DEFAULT_CONNECTION_TIMEOUT_MS = 10000;
-  static final int DEFAULT_REQUEST_TIMEOUT_MS = 60000;
-  static final int DEFAULT_IDLE_TIMEOUT_MS = 4750;
+public final class Connection implements AutoCloseable {
+
+  private static final int DEFAULT_CONNECTION_TIMEOUT_MS = 10000;
+  private static final int DEFAULT_REQUEST_TIMEOUT_MS = 60000;
+  private static final int DEFAULT_IDLE_TIMEOUT_MS = 4750;
+  private static final int DEFAULT_MAX_RETRIES = 0;
+  private static final URL FAUNA_ROOT;
+
+  static {
+    try {
+      FAUNA_ROOT = new URL("https://rest.faunadb.com");
+    } catch (MalformedURLException e) {
+      throw new RuntimeException(e); // won't happen
```

Comment:
Probably use an https://docs.oracle.com/javase/7/docs/api/java/lang/Error.html subclass for this